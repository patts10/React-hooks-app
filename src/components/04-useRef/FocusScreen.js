import React, { useRef } from 'react';
import '../02-useEffect/effects.css'

export const FocusScreen = () => {

  const inputRef = useRef();
  console.log(inputRef);

  const handleClick = () => {
    inputRef.current.select();
  }


  return (
    <>
      <h1>Focus Screen</h1>
      <hr />

      <input
        className="block mr-4 mb-5"
        ref={ inputRef }
        type="text" 
        placeholder="Su nombre" />

      <button
        className="px-6 py-2.5 bg-white text-blue-600 hover:text-white font-medium text-xs border border-blue-600 leading-tight rounded shadow-md hover:bg-blue-700 hover:shadow-lg focus:bg-blue-700 focus:shadow-lg focus:outline-none focus:ring-0 active:bg-blue-800 active:shadow-lg transition duration-150 ease-in-out"
        onClick={ handleClick }
      >
        Focus
      </button>
    </>
  );
}
