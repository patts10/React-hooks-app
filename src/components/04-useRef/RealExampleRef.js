import React, { useState } from 'react';
import '../02-useEffect/effects.css';
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks';

export const RealExampleRef = () => {

  const [show, setShow] = useState(false)

  return (
    <>
      <h1>Real Ref</h1>
      <hr />

      { show && <MultipleCustomHooks />}

      <button
        className='px-6 py-2.5 ml-3 bg-white text-blue-600 hover:text-white font-medium text-xs border border-blue-600 leading-tight rounded shadow-md hover:bg-blue-700 hover:shadow-lg focus:shadow-lg focus:outline-none focus:ring-0 active:bg-blue-800 active:shadow-lg transition duration-150 ease-in-out'
        onClick={ () => {
          setShow(!show);
        }}
      >
        Show/Hide
      </button>
    </>
  )
}
