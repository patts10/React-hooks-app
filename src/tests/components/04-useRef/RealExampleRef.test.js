import { RealExampleRef } from "../../../components/04-useRef/RealExampleRef";
import { shallow } from 'enzyme';

describe('Tests on <RealExampleRef />', () => { 
  
  const wrapper = shallow( <RealExampleRef /> );
  
  test('should show correctly', () => { 
    
    expect(wrapper).toMatchSnapshot();
  });
  
  test('should show <MultipleCustomHooks /> component', () => { 
    
    wrapper.find('button').simulate('click');

    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);

    });
 })