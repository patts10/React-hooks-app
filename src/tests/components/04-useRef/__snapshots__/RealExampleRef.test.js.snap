// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <RealExampleRef /> should show correctly 1`] = `
<Fragment>
  <h1>
    Real Ref
  </h1>
  <hr />
  <button
    className="px-6 py-2.5 ml-3 bg-white text-blue-600 hover:text-white font-medium text-xs border border-blue-600 leading-tight rounded shadow-md hover:bg-blue-700 hover:shadow-lg focus:shadow-lg focus:outline-none focus:ring-0 active:bg-blue-800 active:shadow-lg transition duration-150 ease-in-out"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</Fragment>
`;
